pkg/security/apiserver/registry/securitycontextconstraints/etcd/etcd.go

```diff
@@ -1,55 +1,36 @@
 package etcd
 
 import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apiserver/pkg/registry/generic"
-	"k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/apiserver/pkg/registry/rest"
-	"k8s.io/kubernetes/pkg/printers"
-	printerstorage "k8s.io/kubernetes/pkg/printers/storage"
 
-	"github.com/openshift/api/security"
-	printersinternal "github.com/openshift/origin/pkg/printers/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	securityapi "github.com/openshift/origin/pkg/security/apis/security"
-	"github.com/openshift/origin/pkg/security/apiserver/registry/securitycontextconstraints"
 )
 
 // REST implements a RESTStorage for security context constraints against etcd
 type REST struct {
-	*registry.Store
 }
 
 var _ rest.StandardStorage = &REST{}
 var _ rest.ShortNamesProvider = &REST{}
+var _ rest.Scoper = &REST{}
 
 // ShortNames implements the ShortNamesProvider interface. Returns a list of short names for a resource.
 func (r *REST) ShortNames() []string {
 	return []string{"scc"}
 }
 
 // NewREST returns a RESTStorage object that will work against security context constraints objects.
-func NewREST(optsGetter generic.RESTOptionsGetter) *REST {
-	store := &registry.Store{
-		NewFunc:     func() runtime.Object { return &securityapi.SecurityContextConstraints{} },
-		NewListFunc: func() runtime.Object { return &securityapi.SecurityContextConstraintsList{} },
-		ObjectNameFunc: func(obj runtime.Object) (string, error) {
-			return obj.(*securityapi.SecurityContextConstraints).Name, nil
-		},
-		PredicateFunc:            securitycontextconstraints.Matcher,
-		DefaultQualifiedResource: security.Resource("securitycontextconstraints"),
-
-		TableConvertor: printerstorage.TableConvertor{TablePrinter: printers.NewTablePrinter().With(printersinternal.AddHandlers)},
-
-		CreateStrategy:      securitycontextconstraints.Strategy,
-		UpdateStrategy:      securitycontextconstraints.Strategy,
-		DeleteStrategy:      securitycontextconstraints.Strategy,
-		ReturnDeletedObject: true,
-	}
-	options := &generic.StoreOptions{RESTOptions: optsGetter, AttrFunc: securitycontextconstraints.GetAttrs}
-	if err := store.CompleteWithOptions(options); err != nil {
-		panic(err) // TODO: Propagate error up
-	}
-	return &REST{store}
+func NewREST() *REST {
```

Comment:
move this change to another pull

Reply:
done, moved to same pull as admission plugin changes [here](https://github.com/openshift/origin/pull/22658)